MuxExoPlayer/src/main/java/com/mux/stats/sdk/muxstats/AdsImaSDKListener.java

```diff
@@ -0,0 +1,75 @@
+package com.mux.stats.sdk.muxstats;
+
+import com.google.ads.interactivemedia.v3.api.AdErrorEvent;
+import com.google.ads.interactivemedia.v3.api.AdEvent;
+import com.mux.stats.sdk.core.events.playback.AdBreakEndEvent;
+import com.mux.stats.sdk.core.events.playback.AdBreakStartEvent;
+import com.mux.stats.sdk.core.events.playback.AdEndedEvent;
+import com.mux.stats.sdk.core.events.playback.AdFirstQuartileEvent;
+import com.mux.stats.sdk.core.events.playback.AdMidpointEvent;
+import com.mux.stats.sdk.core.events.playback.AdPauseEvent;
+import com.mux.stats.sdk.core.events.playback.AdPlayEvent;
+import com.mux.stats.sdk.core.events.playback.AdPlayingEvent;
+import com.mux.stats.sdk.core.events.playback.AdRequestEvent;
+import com.mux.stats.sdk.core.events.playback.AdResponseEvent;
+import com.mux.stats.sdk.core.events.playback.AdThirdQuartileEvent;
+
+public class AdsImaSDKListener implements AdErrorEvent.AdErrorListener, AdEvent.AdEventListener {
+    private MuxBaseExoPlayer exoPlayerListener;
+
+    public AdsImaSDKListener(MuxBaseExoPlayer listener) {
+        exoPlayerListener = listener;
+    }
+
+    @Override
+    public void onAdError(AdErrorEvent adErrorEvent) {
+        if (exoPlayerListener != null)
+            exoPlayerListener.dispatch(new com.mux.stats.sdk.core.events.playback.AdErrorEvent(null));
+    }
+
+    @Override
+    public void onAdEvent(AdEvent adEvent) {
+        if (exoPlayerListener != null) {
+            switch (adEvent.getType()) {
+                case AD_BREAK_STARTED:
+                    exoPlayerListener.dispatch(new AdBreakStartEvent(null));
+                    break;
+                case AD_BREAK_ENDED:
+                    exoPlayerListener.dispatch(new AdBreakEndEvent(null));
+                    break;
+                case STARTED:
+                    exoPlayerListener.dispatch(new AdPlayEvent(null));
+                    exoPlayerListener.dispatch(new AdPlayingEvent(null));
```

Comment:
This is a similar shortcut as I had to take in some javascript implementations, but in this world can we get the playhead time or anything more specific on this? Knowing the difference between `play` and `playing` is really important here to be able to distinguish between ad request time, load time, and startup time.

Reply:
For instance, does this event help us out at all here? `public static final AdEvent.AdEventType AD_PROGRESS`

Reply:
fixed.